sgt-back-end/server.ts

```diff
@@ -0,0 +1,151 @@
+import express from 'express';
+import { ClientError } from './client-error.js';
+import { errorMiddleware } from './error-middleware.js';
+import pg from 'pg';
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/api/grades', async (req, res, next) => {
+  const { name, course, score } = req.body;
+
+  if (
+    !name ||
+    !course ||
+    typeof score !== 'number' ||
+    score < 0 ||
+    score > 100
+  ) {
+    return next(
+      new ClientError(
+        400,
+        'Invalid grade. Ensure name, course are provided and score is between 0 and 100.'
+      )
+    );
+  }
+
+  const sql = `
+    INSERT INTO "grades" ("name", "course", "score")
+    VALUES ($1, $2, $3)
+    RETURNING *
+  `;
+  const params = [name, course, score];
+
+  try {
+    const result = await db.query(sql, params);
+    res.status(201).json(result.rows[0]);
+  } catch (err) {
+    next(new ClientError(500, 'Database query failed.'));
+  }
+});
+
+app.get('/api/grades', async (req, res, next) => {
+  const sql = `
+    SELECT *
+    FROM "grades"
+    `;
+
+  try {
+    const result = await db.query(sql);
+    res.json(result.rows);
+  } catch (err) {
+    next(new ClientError(500, 'Database query failed.'));
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  const gradeId = +req.params.gradeId;
```

Comment:
You should check if `gradeId` is NaN. If it is, the select won't select anything and you'll return a 404, but it's best to catch it early and return a 400.